### DISCUSSION TITLE
no-trailing-spaces rule

### DISCUSSION THREAD

**Discussion by leandrocunha** [terraform-linters/tflint#1819] [Q&A] [answered]
Hey guys,
There is any rule like `no-trailing-spaces` as we have in the eslint
https://eslint.org/docs/latest/rules/no-trailing-spaces

I would to highlight extra/wrong spaces like the image below
![image](https://github.com/terraform-linters/tflint/assets/1820920/c72d55b7-8b1c-466d-b25a-78723fe38f5b)

**Comment by bendrucker** [comment, marked as answer]
TFLint does not try to deal with formatting. You should use `terraform fmt -check` for this. 

Terraform, similar to Go, provides a canonical formatter as part of the core tool, to mitigate the sort of stylistic bikeshedding seen in other languages, especially JavaScript. The only exceptions where it's appropriate for TFLint to get involved via a core ruleset is where `terraform fmt` allows multiple syntaxes for compatibility but the official documentation recommends a certain style. There's only one case of that so far:

https://github.com/terraform-linters/tflint-ruleset-terraform/blob/main/docs/rules/terraform_comment_syntax.md

It's a non-goal for the `terraform` ruleset to provide configurable enforcement of personal/organizational style conventions. Of course you're welcome to write custom rulesets for any stylistic preferences you want to enforce in your modules.
